### DISCUSSION TITLE
Android nightly 1.2.4 keyboard presses don't get to the Windows controlled device,

### DISCUSSION THREAD

**Discussion by flashgordonAR** [rustdesk/rustdesk#6417] [Q&A] [answered]
Android nightly 1.2.4 keyboard presses don't get to the Windows controlled device with 1.2.4 nightly.
But if going to Android 1.2.3 stable it worked
Nothing changed from configuration, only upgraded to nightly
Any ideas?
Thanks in advance for any suggestions
Gabriel

**Comment by basilgello** [comment]
Likely Android hardware keyboard support commit issues.

On Android, enable Storage permissions for Rustdesk and reproduce the issue - log files will be in /sdcard/RustDesk/Logs
On Windows, logs should be in %APPDATA%\Rustdesk

**Reply by flashgordonAR** [reply]
Hi basilgello,
  Thanks for the suggestion.  I am running Android 13.
   I tried to get to Rustdesk's permission, thru App Info menu.  But there's no way to add the Storage permissions.
  I have gone thru Permission Manager, selected Files and Media (sounds like Storage permission, as there's no item called Storage),  I see there 2 apps with the permission to access all files on the device.   There is no app with Not allowed.  in the section to see more apps that can access all files, I have searched for Rustdesk and enabled the slider.  Still, I see no Storage or File permission in the app.
  I download each nightly release and check, but no use.  I need to go back to release 1.2.3 stable to be able to input keystrokes from phone to get to Windows device.
  If there's still any log in Windows that can be helpful, let me know the path inside %APPDATA%\Rustdesk directory (there are many .log files), so I can share.  Or, if I have to stick with 1.2.3 stable until 1.2.4 becomes stable, I can wait.
  Thanks again for your help.
Gabriel

**Reply by basilgello** [reply]
@21pages ^^

**Reply by 21pages** [reply]
If not installed, the related log file is `%appdata%\RustDesk\log\RustDesk_rCURRENT.log` 
If installed, the related log file is `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log\server\RustDesk_rCURRENT.log`.
If the `Transfer file` switch is on, android nightly should have logs.
![fed4f2a36f7e6148d4908dfadf7adce](https://github.com/rustdesk/rustdesk/assets/14891774/3528ba10-448d-407f-9182-6fd608aac62e)

**Reply by flashgordonAR** [reply]
[rs_rCURRENT.log](https://github.com/rustdesk/rustdesk/files/13371781/rs_rCURRENT.log)
[RustDesk_rCURRENT.log](https://github.com/rustdesk/rustdesk/files/13371783/RustDesk_rCURRENT.log)
Hi,
  Do these files help?
  Thanks!
Gabriel

**Reply by 21pages** [reply]
Thanks for the log, there seems to be no clue.

**Reply by basilgello** [reply]
I think we need to enable keyboard transferring messages in the log somehow (only the fact of keypress, not the contents of course) and let re-test with a specific build.

**Reply by flashgordonAR** [reply]
Let me know if I have to do something to help.  What build to download, any specific configuration and the logs to send.

**Reply by 21pages** [reply]
I'm not familiar with this part, it seems that the key event is not sent.
https://github.com/21pages/rustdesk/commit/9e933a2711c20715d67e3119b8e8a873a14686af
https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.4-x86_64-android-keymouse-log.exe
https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.4-aarch64-keymouse-log.apk

**Comment by rustdesk** [comment]
confirmed, not working. @fufesou figure out what causes this break ASSP. This is big regression test failure.

**Comment by rustdesk** [comment]
try out the new nightly build please

**Reply by flashgordonAR** [reply]
Hi there, just downloaded the Android nightly build.  Still keystrokes don't get to the Windows machine.  Build 1.2.3 stable works.
I'm not making any changes to any configuration, just installing the .apk

**Reply by rustdesk** [reply]
Sorry for our stupid, I just tested, not working for me either.

**Reply by flashgordonAR** [reply]
Don't worry.  It's ok.  I'm willing to help and test.  It's my collaboration to the project being helpful in some way.

**Reply by rustdesk** [reply, marked as answer]
Please try again, I tested, It is working for me now.

**Reply by flashgordonAR** [reply]
yes! It's working now for me, too.  Thanks a lot for your help and support.
